src/LanguageServer.php

```diff
@@ -69,11 +82,18 @@ public function __construct(ProtocolReader $reader, ProtocolWriter $writer)
      */
     public function initialize(string $rootPath, int $processId, ClientCapabilities $capabilities): InitializeResult
```

Comment:
I wasn't aware `$rootPath` can be null, need to remove the type hint then and adapt the `@param`. Or even better, move it to the end as a parameter and give it default value of `null`. There should also be a test for this.